src/services/identity/SitesService.ts

```diff
@@ -1,46 +1,207 @@
+import { GitHubService } from "@services/db/GitHubService"
+import { ConfigYmlService } from "@services/fileServices/YmlFileServices/ConfigYmlService"
+import IsomerAdminsService from "@services/identity/IsomerAdminsService"
+import UsersService from "@services/identity/UsersService"
 
 import TokenStore from "./TokenStore"
 
 interface SitesServiceProps {
-  repository: ModelStatic<Site>
+  siteRepository: ModelStatic<Site>
+  gitHubService: GitHubService
+  configYmlService: ConfigYmlService
+  usersService: UsersService
+  isomerAdminsService: IsomerAdminsService
   tokenStore: TokenStore
 }
 
 class SitesService {
   // NOTE: Explicitly specifying using keyed properties to ensure
   // that the types are synced.
-  private readonly repository: SitesServiceProps["repository"]
+  private readonly siteRepository: SitesServiceProps["siteRepository"]
+
+  private readonly gitHubService: SitesServiceProps["gitHubService"]
+
+  private readonly configYmlService: SitesServiceProps["configYmlService"]
+
+  private readonly usersService: SitesServiceProps["usersService"]
+
+  private readonly isomerAdminsService: SitesServiceProps["isomerAdminsService"]
 
   private readonly tokenStore: SitesServiceProps["tokenStore"]
 
-  constructor({ repository, tokenStore }: SitesServiceProps) {
-    this.repository = repository
+  constructor({
+    siteRepository,
+    gitHubService,
+    configYmlService,
+    usersService,
+    isomerAdminsService,
+    tokenStore,
+  }: SitesServiceProps) {
+    this.siteRepository = siteRepository
+    this.gitHubService = gitHubService
+    this.configYmlService = configYmlService
+    this.usersService = usersService
+    this.isomerAdminsService = isomerAdminsService
     this.tokenStore = tokenStore
   }
 
   async getBySiteName(siteName: string): Promise<Site | null> {
-    const site = await this.repository.findOne({
+    const site = await this.siteRepository.findOne({
       where: { name: siteName },
     })
 
     return site
   }
 
+  async getSitesForEmailUser(userId: string) {
+    const user = await this.usersService.findSitesByUserId(userId)
+
+    if (!user) {
+      return []
+    }
+
+    return user.site_members.map((site) => site.repo?.name)
+  }
+
+  async getSites(sessionData: UserSessionData): Promise<RepositoryData[]> {
+    const isEmailUser = sessionData.isEmailUser()
+    const { isomerUserId: userId } = sessionData
+    const isAdminUser = !!(await this.isomerAdminsService.getByUserId(userId))
+
+    const { accessToken } = sessionData
+    const endpoint = `https://api.github.com/orgs/${ISOMER_GITHUB_ORG_NAME}/repos`
+
+    // Simultaneously retrieve all isomerpages repos
+    const paramsArr = _.fill(Array(ISOMERPAGES_REPO_PAGE_COUNT), null).map(
+      (_, idx) => ({
+        per_page: GH_MAX_REPO_COUNT,
+        sort: "full_name",
+        page: idx + 1,
+      })
+    )
+
+    const allSites: RepositoryData[] = await Bluebird.map(
+      paramsArr,
+      async (params) => {
```

Comment:
turns out this is typed as `any`... from bluebird [docs](https://github.com/petkaantonov/bluebird), 

> Please use native promises instead if at all possible.

so if it's not a big chore, could we do this using native `map`